extensions/amp-analytics/analytics-vars.md

```diff
@@ -0,0 +1,105 @@
+# <a name="amp-analytics"></a> Variables supported in `amp-analytics`
+
+## Using the `vars`
+
+The format `${var-name}` can be used when a page or platform defined variable needs to be used in a request string. The `amp-analytics` tag will replace the template with the actual value at the time of construction of the analytics request. All the values substituted using this method are first url encoded using `encodeUrlComponent`. For example:
+
+```html
+<amp-analytics>
+<script type="application/json">
+{
+  "requests": {
+    "pageview": "https://example.com/analytics?url=${canonicalUrl}&title=${title}&acct=${account}",
+    ...
+  },
+  "vars": {
+    "account": "ABC123"
+  },
+  "triggers": {
+    ...
+  }
+}
+</script>
+</amp-analytics>
+```
+
+## Supported Vars
+
+**random** provides a random value every time a request is being constructed.
+
+Example value: `0.12345632345`
+
+**canonicalUrl** provides the canonical URL of the current document.
+
+Example value: `http%3A%2F%2Fexample.com%3A8000%2Fanalytics.html`
+
+**canonicalHost** provides the canonical URL's host.
+
+Example value: `example.com`
+
+**canonicalPath** provides the canonical URL's path part.
+
+Example value: `%2Fanalytics.html`
+
+**clientId** provides a per document-source-origin (The origin of the website where you publish your AMP doc) and user identifier. The client id will be the same for the same user if they visit again within one year. If the AMP document is not served through the AMP CDN, the client id will be replaced with a cookie of the name of the cid scope below. If the cookie is not present, an empty string will be returned.
+
+Please see below the required and optional arguments you may pass into `clientId` like a function. (spaces between arguments and values are not allowed)
+
+arguments:
+
+`cid-scope` (Required) - Name of the fallback cookie when the document is not served by an AMP proxy.
+`amp-user-notification` id (Optional) - Optionally make the clientId substitution be dependent on the dismissal of a user notification shown to the visitor of the page.
+
+Example usage: `${clientId(foo)}`
+
+Example value: `U6XEpUs3yaeQyR2DKATQH1pTZ6kg140fvuLbtl5nynbUWtIodJxP5TEIYBic4qcV`
+
+
+**documentReferrer** provides the referrer where the user came from. It is read from `document.referrer`. The value is empty for direct visitors.
+
+Example value: `https://www.google.com`
+
+**title** provides the title of the current document.
+
+Example value: `Example Pageview`
+
+**ampdocUrl** provides the AMP document's URL. The url contains the scheme, domain, port and full path. It does not contian the fragment part of the URL.
+
+Example value: `http://example.com:8000/examples.build/analytics.amp.max.html`
+
+**ampdocHost** provides the AMP document's URL host.
+
+Example value: `example.com`
+
+**pageViewId** provides a string that is intended to be random and likely to be unique per URL, user and day.
+
+Example value: `978`
```

Comment:
The values go up to 10000. Most of the values generated here from my local machine :). I didn't want to mention the limit because that is subject to change.